README.md

```diff
@@ -172,6 +161,40 @@ ECR registry:
 This is useful if you use `docker` to operate on registries that use different
 authentication credentials.
 
+### AWS credentials
+
+The Amazon ECR Docker Credential Helper allows you to use AWS credentials stored in different locations.  Standard ones
+include:
+
+* The shared credentials file (`~/.aws/credentials`)
+* The `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` environment variables
+* An [IAM role for an Amazon ECS task](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-iam-roles.html)
+* An [IAM role for Amazon EC2](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html)
+
+To use credentials associated with a different named profile in the shared credentials file (`~/.aws/credentials`), you
+may set the `AWS_PROFILE` environment variable. 
+
+The Amazon ECR Docker Credential Helper can optionally read and support some configuration options specified in the AWS
+shared configuration file (`~/.aws/config`).  To use these options, you must set the `AWS_SDK_LOAD_CONFIG` environment
+variable to `true`.  The supported options include:
+
+* Assumed roles specified with `role_arn` and `source_profile`
+* External credential processes specified with `credential_process`
+* Web Identities like [IAM Roles for Service Accounts in
+  Kubernetes](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) (*Note: Kubernetes
+  users using containers with a non-root user may encounter permission issues described in [this
+  bug](https://github.com/kubernetes-sigs/external-dns/pull/1185) and may need to employ a workaround adjusting the
+  Kubernetes `securityContext`.*)
+
+The Amazon ECR Docker Credential Helper uses the same credentials as the AWS
```

Comment:
This covers the credentials being used with the same precedence as the aws cli?

Reply:
Roughly yes, but the precise statement is that the credentials are used with the same precedence as the AWS SDK for Go.  The SDK attempts to match the CLI, but is sometimes imperfect at doing so (and some of the changes in this release are bugfixes for that).

Reply:
If we would consider it a bug if we don't match the CLI I think it's fine. If not, is it worth pointing out any known differences and/or pointing to the "Specifying Credentails" section here? https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html.

Reply:
The CLI docs are more "user-facing" than the SDK docs since the CLI covers an application (the SDK docs cover how to write code using the SDK as well).  One of the major known differences is already pointed out in this section, specifically the part about `AWS_SDK_LOAD_CONFIG` being required when any value is to be read from the shared config file (`~/.aws/config`).

If you don't feel strongly about this, maybe we can move this to a separate PR/issue after this release goes out?